pkgs/common-updater/scripts/update-source-version

```diff
@@ -229,10 +229,11 @@ if [[ -n "$newUrl" ]]; then
         die "Couldn't evaluate source url from '$attr.$sourceKey'!"
     fi
 
-    # Escape regex metacharacter that are allowed in store path names
-    oldUrlEscaped=$(echo "$oldUrl" | sed -re 's|[${}.+?]|\\&|g')
+    # Escape regex metacharacter that may appear in URLs
+    oldUrlEscaped=$(echo "$oldUrl" | sed -re 's|[${}.+?\[]|\\&|g')
+    newUrlEscaped=$(echo "$newUrl" | sed -re 's|&|\\&|g')
 
-    sed -i.cmp "$nixFile" -re "s|\"$oldUrlEscaped\"|\"$newUrl\"|"
+    sed -i.cmp "$nixFile" -re "s|\"$oldUrlEscaped\"|\"$newUrlEscaped\"|"
```

Comment:
Switching to BRE so that we do not need to escape `+{?`:

```suggestion
    sed -i.cmp "$nixFile" -e "s|\"$oldUrlEscaped\"|\"$newUrlEscaped\"|"
```